### PR TITLE
re:Invent 2020 features

### PR DESCRIPTION
**Pull request by danabens**
re:Invent 2020 features

* log graph based artifacts for rendering in Studio such as ROC Curves
* various code quality improvements

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -411,19 +428,26 @@ def docker_image(boto_model_file, sagemaker_endpoint):
...
 
-    os.makedirs("tests/integ-jobs/docker/boto", exist_ok=True)
+    shutil.copy(sdist_path, os.path.join(docker_dir, "smexperiments-0.1.0.tar.gz"))
 
+    # may need to configure cred helper in ~/.docker/config.json
+    # {
+    # 	"credHelpers": {
+    # 		"aws_account_id.dkr.ecr.region.amazonaws.com": "ecr-login"
+    # 	}
+    # }
```
Comment (yzhu0) [inline tests/conftest.py:448]: do we need these?

File: tests/unit/artifact_schemas/precision_recall_curve_v0.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      "description": "Decreasing recall values such that element i is the recall of predictions with score >= thresholds[i] and the last element is 0.",
+      "type": "array"
+    },
+    "averagePrecisionScore": {
+      "description": "AP summarizes a precision-recall curve as the weighted mean of precisions achieved at each threshold, with the increase in recall from the previous threshold used as the weight.",
+      "type": "number"
+    }
+  },
+  "additionalProperties": false
+}
```
Comment (yzhu0) [inline tests/unit/artifact_schemas/precision_recall_curve_v0.json:34]: nit: add new line at the end of file, same as a few class below.

**Review by yzhu0 — APPROVED** [review]

File: tests/conftest.py (hunk 1 above)
Comment (danabens) [reply tests/conftest.py:448]: ya this is a prerequisite setup step to get this test to work

File: tests/unit/artifact_schemas/precision_recall_curve_v0.json (hunk 2 above)
Comment (danabens) [reply tests/unit/artifact_schemas/precision_recall_curve_v0.json:34]: added

**Pull request merged**